tokenserver/tests/test_service.py

```diff
@@ -791,6 +791,31 @@ def test_node_type_is_returned_in_response(self):
         res = self.app.get('/1.0/sync/1.1', headers=headers, status=200)
         self.assertEqual(res.json['node_type'], 'example')
 
+    def test_migrate_new_user(self):
+        EMAIL0 = "abO-test@example.com"
+        EMAIL1 = "abT-test@example.com"
+        settings = self.backend.settings
+        # the two platfroms differ in how the settings are labeled.
```

Comment:
Per my comment on the memory-node backend, I don't think this should be necessary. If you try to make the memory backend access the settings through `**kw` in the same way as the SQL backend, what fails?

Reply:
It's not a full on fail, just a mis-match between the various ways that these values are set and read. I've added a `settings` property to try and normalize the values that are being passed in regardless of how. For what it's worth, it also gave me a cleaner way to do test settings.

Reply:
> It's not a full on fail, just a mis-match between the various ways that these values are set
> and read.

I guess I don't understand why these values are getting set in more than one way; every creation of an `INodeAssignment` backend should be being routed through the call to `load_and_register("tokenserver", config)` in `tokenserver/__init__.py`, but maybe that's not happening in all cases.

Reply:
Also, removed this comment, since the setting normalize function invalidates it.

Reply:
Ok, having dug into this a bit more:
The problem is that there are two different pools from which "settings" can be derived, and they overlap a bit. During the init, `kw` values are extracted from the pyramid settings and the various plugins are called with those `kw` values, trimmed of their prefixes. Then pyramid creates some thread local values that are the original settings values with prefixes. 

The tokenserver code [sometimes calls the thread local settings](https://github.com/mozilla-services/tokenserver/blob/daf36fb991e0b3fe34dd21d62c20d192f968ff44/tokenserver/assignment/memorynode.py#L28), which as I understand, could change dynamically. Mind you, this sort of call doesn't happen regularly, and the one more problematic one I highlighted earlier probably could be converted to a `kw` init to be consistent with other forms. (There's another time the code [calls to the thread local settings](https://github.com/mozilla-services/tokenserver/blob/2ade4a7c6ab8658a11a7674c47ea74afdc7ad2bb/tokenserver/verifiers.py#L32) registry, but that one appears to be unrelated to configuration.

If the assignment node configurations aren't as dynamic, I can simplify this code a good deal, remove the thread local transmogrification bits, etc.